### PR TITLE
Reader endianness

### PR DESCRIPTION
**Pull request by ghost**
Swaps the `Reader` and `BinaryObject` class endianness methods to use `Constants::Endianness` enum class.

### CODE REVIEW

File: src/BinaryObject.h [hunk 1]
```diff
@@ -26,7 +27,9 @@ class BinaryObject {
   void setCpuSubType(CpuType type);
 
   [[nodiscard]] bool isLittleEndian() const;
-  void setLittleEndian(bool little);
+  [[nodiscard]] bool isBigEndian() const;
+
+  void setEndianness(Constants::Endianness endianness);
```
Comment (netromdk) [inline src/BinaryObject.h:32]: And changing into having only the two set+get endianness functions here, like in Reader, too.

File: src/formats/MachO.cc [hunk 2]
```diff
@@ -2,8 +2,10 @@
 #include <QFile>
 
 #include <cmath>
+#include <iterator>
```
Comment (netromdk) [inline src/formats/MachO.cc:5]: Why is this added exactly?

**Review by netromdk — CHANGES_REQUESTED** [review]
Really great changes and it's much easier to read with the enum instead of bool for Endianness! 👍🏻

File: src/formats/MachO.cc (hunk 2 above)
Comment (0verbyte) [reply src/formats/MachO.cc]: I think it was a mistake automatic import. Removed.

**Comment by 0verbyte** [comment]
Changes have been pushed. I also better organized the commits.

**Review by netromdk — APPROVED** [review]
Thanks a lot!
